llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -92,6 +96,36 @@ static cl::opt<bool> SplitEsimd{"split-esimd",
                                 cl::desc("Split SYCL and ESIMD kernels"),
                                 cl::cat(PostLinkCat)};
 
+static cl::opt<bool> LowerEsimd{
```

Comment:
```suggestion
// TODO Design note: sycl-post-link should probably separate different kinds of its
// functionality on  logical and source level:
//  - LLVM IR module splitting
//  - Running LLVM IR passes on resulting modules
//  - Generating additional files (like spec constants, dead arg info,...)
// The tool itself could be just a "driver" creating needed pipelines from the above actions.
// This could help make the tool structure clearer and more maintainable.


implement only those transformations all pass-based IR t-forms should be done using either opt invoked directly from the driver or using some custom pass manager separated from sycl-post-link.

static cl::opt<bool> LowerEsimd{
```